esmvaltool/cmorizers/data/formatters/datasets/esacci_seaice.py

```diff
@@ -0,0 +1,319 @@
+"""ESMValTool CMORizer for ESACCI-SEAICE data.
+
+Tier
+   Tier 2: other freely-available dataset.
+
+Source
+   ftp://anon-ftp.ceda.ac.uk/neodc/esacci/sea_ice/data
+
+Last access
+   20241107
+
+Download and processing instructions
+   Download the data from:
+       sea_ice_concentration/L4/ssmi_ssmis/12.5km/v3.0/{year}/{month}
+   Put all files under a single directory (no subdirectories
+   with years or months).
+"""
+import glob
+import logging
+import os
+from copy import deepcopy
+from datetime import datetime
+from dateutil import relativedelta
+
+import cf_units
+import iris
+import numpy as np
+from dask import array as da
+from esmvalcore.cmor._fixes.common import OceanFixGrid
+from esmvalcore.cmor.table import CMOR_TABLES
+from esmvalcore.preprocessor import monthly_statistics
+from iris.coords import AuxCoord
+from esmvaltool.cmorizers.data import utilities as utils
+
+from ...utilities import save_variable
+
+logger = logging.getLogger(__name__)
+
+
+def _create_nan_cube(cube, year, month, day, is_daily):
+    """Create cube containing only nan from existing cube."""
+    nan_cube = cube.copy()
+    nan_cube.data = da.ma.masked_greater(cube.core_data(), -1e20)
+
+    # Read dataset time unit and calendar from file
+    dataset_time_unit = str(nan_cube.coord('time').units)
+    dataset_time_calender = nan_cube.coord('time').units.calendar
+    # Convert datetime
+    if is_daily:
+        hrs = 12
+    else:
+        hrs = 0
+    newtime = datetime(year=year, month=month, day=day,
+                       hour=hrs, minute=0, second=0, microsecond=0)
+    newtime_num = cf_units.date2num(newtime, dataset_time_unit,
+                                    dataset_time_calender)
+    nan_cube.coord('time').points = float(newtime_num)
+
+    # remove existing time bounds and create new bounds
+    coord = nan_cube.coord('time')
+    if is_daily:
+        bnd1 = newtime + relativedelta.relativedelta(hours=-12)
+        bnd2 = bnd1 + relativedelta.relativedelta(days=1)
+    else:
+        bnd1 = newtime + relativedelta.relativedelta(days=-day + 1)
+        bnd2 = bnd1 + relativedelta.relativedelta(months=1)
+    coord.bounds = [cf_units.date2num(bnd1, dataset_time_unit,
+                                      dataset_time_calender),
+                    cf_units.date2num(bnd2, dataset_time_unit,
+                                      dataset_time_calender)]
+
+    return nan_cube
```

Comment:
The raw data is daily, do we need to keep the code for monthly data??

Reply:
Yes, we do want to output monthly means as we mostly work with this frequency (e.g. in REF).